index.js

```diff
@@ -136,36 +191,49 @@ function nowServing(line) {
 }
 
 const line = []
+var ticketNum = 0
 
 // ***** Scope & Closures - Question 1 *****
+// *** Uncomment the lines below to test
+console.log("%cScope & Closures - Question 1", "color: red")
 
+function takeATicketNumber (line) {
 
-// *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+  ticketNum += 1
+  return `Welcome. You are ticket number ${ticketNum}.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+}
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(nowServing(line))
+// => `Currently serving 2.`
+
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
+
+const newLine = []
+const takeATicketNumberFunc = ticketNumberGeneratorFunc()
+
+function ticketNumberGeneratorFunc(newLine) {
+
+  
+}
```

Comment:
Can be solved by:

```
function ticketNumberGeneratorFunc() {
  let number = 0;
  return function (line) {
    number += 1;
    line.push(number);
    return `Welcome. You are ticket number ${number}`;
  }
}
```

Basically all our logic is encapsulated in this function so we can pass in the argument of any line and then run all the logic of the first deliverable